src/TimeoutMigrationTool/RabbitMq/RabbitMqTimeoutCreator.cs

```diff
@@ -36,8 +36,30 @@ public Task CompleteBatch(int number)
 
         Task<MigrationCheckResult> VerifyEndpointIsReadyForNativeTimeouts(EndpointInfo endpoint)
         {
+            var result = new MigrationCheckResult();
+            if ((endpoint.LongestTimeout - DateTime.UtcNow).TotalSeconds > RabbitBatchWriter.MaxDelayInSeconds)
+            {
+                result.Problems.Add($"{endpoint.EndpointName} - has a timeout that has further away date than allowed {RabbitBatchWriter.MaxDelayInSeconds} seconds.");
+            }
+
+            using (var connection = factory.CreateConnection())
+            using (var channel = connection.CreateModel())
+            {
+                try
+                {
+                    channel.QueueDeclarePassive(endpoint.EndpointName);
+                }
+                catch (Exception)
+                {
+                    result.Problems.Add($"There is no queue for an endpoint {endpoint.EndpointName}.");
+                    return Task.FromResult(result);
+                }
+
+                channel.ExchangeBind(endpoint.EndpointName, "nsb.delay-delivery", $"#.{endpoint.EndpointName}");
```

Comment:
Is there any way to know if this resulted in the binding being created?

Reply:
if the call fails it will throw, but I am hesitant if we want to catch it here. As we know that the queue exists, and the exchange should exist as well. So if something fails it is problem with broker.